internal/engine/postgresql/parse_default.go

```diff
@@ -0,0 +1,38 @@
+//go:build !windows && cgo
+// +build !windows,cgo
+
+package postgresql
+
+import (
+	cgonodes "github.com/pganalyze/pg_query_go/v4"
+	wasinodes "github.com/wasilibs/go-pgquery"
+	"google.golang.org/protobuf/proto"
+)
+
+func parseNodes(input string) (*wasinodes.ParseResult, error) {
+	cgoRes, err := cgonodes.Parse(input)
+	if err != nil {
+		return nil, err
+	}
+
+	// It would be too tedious to maintain conversion logic in a way that
+	// can target the two types from different packages despite being identical.
+	// We go ahead and take a small performance hit by marshaling through
+	// protobuf to unify the types. We must use the wasilibs version because
+	// the upstream version requires cgo even when only accessing the proto.
+
+	resBytes, err := proto.Marshal(cgoRes)
```

Comment:
This is the performance regression, protobuf marshaling is very fast and it's actually relatively common to do this sort of conversion in real code I've seen. But I will try to get go-pgquery to use the same parsetree type as upstream to be able to avoid it, hopefully that goes smoothly